Sources/Common/Core/Math/index.js

```diff
@@ -1995,6 +2044,93 @@ export function solve3PointCircle(p1, p2, p3, center) {
   return radius;
 }
 
+export function computeHistogram(imageData, worldBounds, voxelFunc = null) {
+  if (!imageData) {
+    return {
+      min: 0,
+      max: 0,
+      average: 0,
+    };
+  }
+
+  const bounds = [0, 0, 0, 0, 0, 0];
+  imageData.worldToIndexBounds(worldBounds, bounds);
+
+  const point1 = [0, 0, 0];
+  const point2 = [0, 0, 0];
+  computeBoundsPoints(point1, point2, bounds);
+
+  const dimensions = imageData.getDimensions();
+
+  clampVector(
+    point1,
+    point1,
+    [0, 0, 0],
+    [dimensions[0] - 1, dimensions[1] - 1, dimensions[2] - 1]
+  );
+  clampVector(
+    point2,
+    point2,
+    [0, 0, 0],
+    [dimensions[0] - 1, dimensions[1] - 1, dimensions[2] - 1]
+  );
+
+  roundVector(point1, point1);
+  roundVector(point2, point2);
+
+  const ystride = dimensions[0];
+  const zstride = dimensions[0] * dimensions[1];
+
+  const pixels = imageData
+    .getPointData()
+    .getScalars()
+    .getData();
+  let imax = -Infinity;
+  let imin = Infinity;
+  let isum = 0;
+  let inum = 0;
+
+  for (let z = point1[2]; z <= point2[2]; z++) {
+    for (let y = point1[1]; y <= point2[1]; y++) {
+      for (let x = point1[0]; x <= point2[0]; x++) {
+        if (!voxelFunc || voxelFunc([x, y, z], bounds)) {
+          const index = x + y * ystride + z * zstride;
+          const pixel = pixels[index];
+
+          if (pixel > imax) imax = pixel;
+          if (pixel < imin) imin = pixel;
+          isum += pixel;
```

Comment:
be careful, if you have many pixels, you might overflow the capacity of isum.

Reply:
To overflow a 64 bit number this way, if all pixels are at maximum intensity, you would need a 268435456 by 268435456 pixels image.

Reply:
Good point, I keep forgetting differences between JS and  C++